lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -54,7 +59,7 @@ def discover
         node = @zk.get("#{@discovery['path']}/#{id}")
 
         begin
-          host, port, name = deserialize_service_instance(node.first)
+          host, port, name, weight = deserialize_service_instance(node.first)
```

Comment:
i'm guessing that you're using `weight` to preferentially route within latency zones, but i don't see this attribute used anywhere else. is this branch still missing some code?